### PR TITLE
General housekeeping

### PR DESCRIPTION
**Pull request by sandhose**
I've been bothered by the warnings telling us about the deprecation of the node16 runner.

This does three things:

- Make the action run on node20
- Update ncc and @actions/core
- Switch the source file to use TypeScript

This is reviewable commit by commit.

### CODE REVIEW

File: index.ts
Comment (anoadragon453) [inline index.ts:1]: nit: While I like two-space indentation, I think we generally use four-space for TypeScript across matrix-org repos.

e.g. https://github.com/matrix-org/matrix-react-sdk/blob/2702f8ba8c7f9da77025d1693ac0e6f1229d105a/src/settings/SettingsStore.ts

**Review by anoadragon453 — APPROVED** [review]
LGTM, thanks for doing this!

File: index.ts
Comment (sandhose) [reply index.ts:1]: I added a .editorconfig and reformatted with prettier (which respects whatever is defined in the editorconfig)

Comment (anoadragon453) [reply index.ts:1]: Oh, excellent. Thanks!

**Pull request merged**